File: back-end/pydantec_schemas/enrollment.py
from pydantic import BaseModel
from datetime import datetime
from .student import Student


class EnrollmentBase(BaseModel):
    pass


class EnrollmentCreate(EnrollmentBase):
    student_id: int
    class_id: int


class Enrollment(EnrollmentBase):
    enrollment_id: int
    class_id: int
    student: Student
    created_at: datetime
    updated_at: datetime

    class config:
        orm_mode: True

File: front-end/src/types/enrollment.ts
import { Student } from "./students";

export type EnrollmentClass = {
  enrollment_id: number;
  class_id: number;
  student: Student;
  created_at: string;
  updated_at: string;
};
export type StudentEnrollmentForm = {
  student_id: number;
  class_id: number;
};
